docker_test.go

```diff
@@ -705,6 +705,40 @@ func Test_BuildContainerFromDockerfileWithBuildLog(t *testing.T) {
 	assert.Regexpf(t, `^Step\s*1/\d+\s*:\s*FROM alpine$`, temp[0], "Expected stdout first line to be %s. Got '%s'.", "Step 1/* : FROM alpine", temp[0])
 }
 
+func Test_BuildContainerFromDockerfileWithBuildLogWriter(t *testing.T) {
+	var buffer bytes.Buffer
+
+	ctx := context.Background()
+
+	// fromDockerfile {
+	req := ContainerRequest{
+		FromDockerfile: FromDockerfile{
+			Context:        filepath.Join(".", "testdata"),
+			Dockerfile:     "buildlog.Dockerfile",
+			PrintBuildLog:  true,
+			BuildLogWriter: &buffer,
+		},
+	}
+	// }
+
+	genContainerReq := GenericContainerRequest{
+		ProviderType:     providerType,
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	c, err := GenericContainer(ctx, genContainerReq)
+	CleanupContainer(t, c)
+	require.NoError(t, err)
+
+	out, err := io.ReadAll(&buffer)
+	require.NoError(t, err)
+
+	temp := strings.Split(string(out), "\n")
+	require.NotEmpty(t, temp)
+	assert.Regexpf(t, `^Step\s*1/\d+\s*:\s*FROM alpine$`, temp[0], "Expected stdout first line to be %s. Got '%s'.", "Step 1/* : FROM alpine", temp[0])
```

Comment:
suggestion: use require instead of assert.

Reply:
This behavior is taken from a previous similar test, if correct it, probably need to correct both. It may be worth leaving it in its current form, because the output format is not the most stable thing

Reply:
Yer we're slowly moving to all require, no need to fix he other instances just don't want to introduce new uses of assert.